lib/kuikka_website/web/controllers/page/forum_controller.ex

```diff
@@ -1,19 +1,65 @@
 defmodule KuikkaWebsite.Web.Page.ForumController do
   use KuikkaWebsite.Web, :controller
+  import Ecto.Query
+
+  alias KuikkaWebsite.Forum.Topic
+  alias KuikkaWebsite.Repo
 
   @spec index(Plug.Conn.t, map) :: Plug.Conn.t
+  def index(conn, %{"view" => "new", "title" => title}) do
+    conn
+    |> assign(:changeset, Topic.changeset(%Topic{}, %{title: title}))
+    |> render("new.html")
+  end
+  def index(conn, %{"view" => "new"}) do
+    conn
+    |> assign(:changeset, Topic.changeset(%Topic{}))
+    |> render("new.html")
+  end
   def index(conn, _params) do
     render conn, "index.html"
   end
 
   @spec show(Plug.Conn.t, map) :: Plug.Conn.t
-  def show(conn, _params) do
-    render conn, "show.html"
+  def show(conn, %{"id" => title, "view" => "edit"}) do
+    Topic
+    |> where([t], t.title == ^title)
+    |> Repo.one()
+    |> case do
+      nil ->
+        redirect(conn, to: forum_path(conn, :index, %{view: "new", title: title}))
```

Comment:
Redirecting to forum list should be more preferred and just putting flash message that forum page was not found